### PR TITLE
[added] Prop for adding aria-hidden to application element

### PR DESCRIPTION
**Pull request by claydiffrient**
This makes it so that when provided an selector to the
ariaHideElement it will augment that element with the aria-hidden
attribute effectively hiding everything except the tray from
screenreaders.

### CODE REVIEW

File: lib/components/Tray.js [hunk 1]
```diff
@@ -13,7 +13,8 @@ export default React.createClass({
     closeTimeoutMS: React.PropTypes.number,
     closeOnBlur: React.PropTypes.bool,
     maintainFocus: React.PropTypes.bool,
-    elementToFocus: React.PropTypes.string
+    elementToFocus: React.PropTypes.string,
+    ariaHideElement: React.PropTypes.string
```
Comment (ryanmt) [inline lib/components/Tray.js]: Why not `isRequired`? Doesn't log in production builds, but it would do a lot for you.

Comment (junyper) [inline lib/components/Tray.js]: I still think having these as functions that return an element would be better ;)

File: lib/helpers/customPropTypes.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// Adapted from https://github.com/react-bootstrap/react-prop-types/blob/master/src/isRequiredForA11y.js
+export function a11yFunction(props, propName, componentName) {
+  if (!props[propName]) {
```
Comment (junyper) [inline lib/helpers/customPropTypes.js]: maybe also check if it's a function?

Comment (claydiffrient) [inline lib/helpers/customPropTypes.js]: Oh wow... that's money... on it!

**Comment by claydiffrient** [comment]
@junyper How's that?

**Pull request merged**